### ISSUE TITLE
Type Narrowing Incorrect on Exported Variable

### ISSUE THREAD

**Issue by ChrisMeyer7088** [microsoft/TypeScript#57817]
### 🔎 Search Terms

Narrowing Export

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about 5.0 and higher releases

### ⏯ Playground Link

https://www.typescriptlang.org/play?#code/DYUwLgBARsCGAWAuCBnMAnAlgOwOYQB8IBXbAExADMcQyBuAKAc0ogAoBCGBASgjHjoA9gHcI2EGICi6YejYByACrxMKCCFlD0EMkJDrsQyNlhaxYAJ4AHEAp5MQAD2vbIAbwYRoceBAD0-hAASuDE6NjqVraoGDj4RKQU1BJkADSo8ELEwGTQIOJmwiK0-EKxWHgMAL5AA

### 💻 Code

```ts
let blah: string | undefined;

if (!blah) throw new Error('This error does not narrow type')

export {
  blah // Returns type string | undefined, should be narrowed to string
}
```

### 🙁 Actual behavior

Expected the throwing of an error to type narrow the exported variable, but it instead held onto its initial typing.

### 🙂 Expected behavior

Throwing an error should type narrow exported variables

**Comment by RyanCavanaugh** [comment]
This is the correct behavior. It's possible to observe `blah` before the `if`, e.g.:
```ts
// self-import in foo.mjs
import { blah as self_blah } from "./foo.mjs";
let blah;
// prints "undefined"
console.log(self_blah);
export {
  blah
}
```
